lib/providers/portion/PortionProvider.ts

```diff
@@ -39,6 +38,33 @@ export class PortionProvider implements IPortionProvider {
     return tokenOutCurrencyAmount.multiply(new Fraction(portion.bips, 10000));
   }
 
+  getPortionQuoteAmount(
+    portionAmount: CurrencyAmount<Currency>,
+    quoteAmount: CurrencyAmount<Currency>,
+    swapperRequestedExactOutAmount: CurrencyAmount<Currency>
+  ): CurrencyAmount<Currency> {
+    // 1. we know the portion amount for exact out with 100% correctness,
+    //    so we can add the portion amount into the exact out amount swapper requested.
+    //    i.e. portionAdjustedAmount = amount + portionAmountToken
+    const portionAdjustedAmount = swapperRequestedExactOutAmount.add(portionAmount);
+    // 2. then we know portion amount and portion adjusted exact out amount,
+    //    we can get a ratio
+    //    i.e. portionToPortionAdjustedAmountSpotPrice = portionAmountToken / portionAdjustedAmount
+    const portionToPortionAdjustedAmountSpotPrice = new Fraction(
```

Comment:
I still think `portionToPortionAdjustedAmountSpotPrice` is a bit wordy / hard to follow. But also get that it's tricky to describe.